src/common/hashes/hash-prefix.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Prefix for hashing functions.
+ *
+ * These prefixes are inserted before the source material used to
+ * generate various hashes. This is done to put each hash in its own
+ * "space." This way, two different types of objects with the
+ * same binary data will produce different hashes.
+ *
+ * Each prefix is a 4-byte value with the last byte set to zero
+ * and the first three bytes formed from the ASCII equivalent of
+ * some arbitrary string. For example "TXN".
+ */
+
+ enum HashPrefix {
+  // transaction plus signature to give transaction ID
+  TX_ID = 0x54584E00, // 'TXN'
```

Comment:
A minor point, but this PR mixes abbreviations and words. Sometimes you refer to variables as `transaction` or `transactionHash` but you'll also refer to them as `tx` or `txHash`.  It would be great to standardize them. 

I'd lean towards not using abbreviations, but I don't have too strong of an opinion.

Reply:
This can be optional - I know that we're primarily just moving code here.